android/src/main/java/com/tailscale/ipn/ui/view/MainView.kt

```diff
@@ -218,47 +217,52 @@ fun PeerList(searchTerm: StateFlow<String>, peers: StateFlow<List<PeerSet>>, onN
             colors = SearchBarDefaults.colors(),
             modifier = Modifier.fillMaxWidth()) {
 
-        Column(
+        LazyColumn(
                 modifier =
                 Modifier
                         .fillMaxSize()
-                        .verticalScroll(rememberScrollState())
                         .background(MaterialTheme.colorScheme.secondaryContainer),
         ) {
             peerList.value.forEach { peerSet ->
-                ListItem(headlineContent = {
-                    Text(text = peerSet.user?.DisplayName
-                            ?: stringResource(id = R.string.unknown_user), style = MaterialTheme.typography.titleLarge)
-                })
+                item {
+                    ListItem(headlineContent = {
+                        Text(text = peerSet.user?.DisplayName
+                                ?: stringResource(id = R.string.unknown_user), style = MaterialTheme.typography.titleLarge)
+                    })
+                }
                 peerSet.peers.forEach { peer ->
-                    ListItem(
-                            modifier = Modifier.clickable {
-                                onNavigateToPeerDetails(peer)
-                            },
-                            headlineContent = {
-                                Row(verticalAlignment = Alignment.CenterVertically) {
-                                    val color: Color = if (peer.Online ?: false) {
-                                        Color.Green
-                                    } else {
-                                        Color.Gray
+                    item {
+                        ListItem(
+                                modifier = Modifier.clickable {
+                                    onNavigateToPeerDetails(peer)
+                                },
+                                headlineContent = {
+                                    Row(verticalAlignment = Alignment.CenterVertically) {
+                                        // By definition, SelfPeer is online since we will not show the peer list unless you're connected.
```

Comment:
Technically you can be offline, but you won't know it by netmap - if you can't reach control, but you had previously managed to get a netmap and are locally "up"